### PR TITLE
[Intents] Update to Xcode8 Beta 4

### PR DESCRIPTION
**Pull request by dalexsoto**
* TODO: We need to bind NSMeasurement<T> in order to enable some members

### CODE REVIEW

**Comment by xamarin-release-manager** [comment]
Build success

File: src/intents.cs [hunk 1]
```diff
@@ -1174,6 +1154,25 @@ interface INDoubleResolutionResult {
 
 	[Introduced (PlatformName.iOS, 10, 0)]
 	[Unavailable (PlatformName.MacOSX)]
+	[DisableDefaultCtor]
+	[BaseType (typeof (INIntentResolutionResult))]
+	interface INDateComponentsResolutionResult {
+
+		[Static]
+		[Export ("successWithResolvedDateComponents:")]
+		INDateComponentsResolutionResult GetSuccess (NSDateComponents resolvedDateComponents);
```
Comment (VincentDondain) [inline src/intents.cs]: Suggestion for `GetSuccess`, `GetDisambiguation` and `GetConfirmationRequired`.

What if, for the parameters, we go:

- `resolvedDateComponents` -> `resolved`
- `componentsToDisambiguate ` -> `toDisambiguate`
- `componentsToConfirm` -> `toConfirm`

File: src/intents.cs [hunk 2]
```diff
@@ -2245,25 +2232,21 @@ interface INRestaurantGuestDisplayPreferences : NSSecureCoding, NSCopying {
...
-		[Export ("contactMethod")]
-		INRestaurantMarketingPreferencesContactMethod ContactMethod { get; }
+	[BaseType (typeof (INIntentResolutionResult))]
+	interface INRestaurantGuestResolutionResult {
 
-		[Export ("guestHasOptedInToMarketing")]
-		bool GuestHasOptedInToMarketing { get; set; }
+		[Static]
+		[Export ("successWithResolvedRestaurantGuest:")]
+		INRestaurantGuestResolutionResult GetSuccess (INRestaurantGuest resolvedRestaurantGuest);
```
Comment (VincentDondain) [inline src/intents.cs]: If https://github.com/xamarin/xamarin-macios/pull/555/files#r73679252 turns out to be the way to go, we should apply the same thing for the 3 methods here.

File: src/intents.cs [hunk 3]
```diff
@@ -2357,6 +2340,25 @@ interface INRestaurantReservationUserBooking : NSCopying {
 
 	[Introduced (PlatformName.iOS, 10, 0)]
 	[Unavailable (PlatformName.MacOSX)]
+	[DisableDefaultCtor]
+	[BaseType (typeof (INIntentResolutionResult))]
+	interface INRestaurantResolutionResult {
+
+		[Static]
+		[Export ("successWithResolvedRestaurant:")]
+		INRestaurantResolutionResult GetSuccess (INRestaurant resolvedRestaurant);
```
Comment (VincentDondain) [inline src/intents.cs]: Same as before: "If https://github.com/xamarin/xamarin-macios/pull/555/files#r73679252 turns out to be the way to go, we should apply the same thing for the 3 methods here."

**Comment by spouliot** [comment]
note: I have `NSMeasurement` bound locally (but Foundation is not complete). I'll look into PR'ing another part of Foundation that contains it asap

File: src/intents.cs [hunk 4]
```diff
@@ -212,8 +212,7 @@ public enum INIntentHandlingStatus : nint {
 		InProgress,
 		Success,
 		Failure,
-		DeferredToApplication,
-		Done = Success
+		DeferredToApplication
```
Comment (spouliot) [inline src/intents.cs]: minor: in the future add `,` on the last enum items. It's legal (in C#) and makes smaller diffs when adding members.

File: src/intents.cs (hunk 1 above)
Comment (spouliot) [inline src/intents.cs]: In general we keep the same names as Apple, unless they are incorrect (wrong/typos) or very confusing. It's globally more consistent (e.g. newer API In future SDK will likely use the names, but the _binder_ might not know we changed them in older API).

**Comment by spouliot** [comment]
LGTM :)

**Pull request merged**